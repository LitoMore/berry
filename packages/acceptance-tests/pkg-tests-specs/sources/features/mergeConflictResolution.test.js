const {
  exec: {execFile},
  fs: {readFile, writeJson},
} = require('pkg-tests-core');

describe(`Features`, () => {
  describe(`Merge Conflict Resolution`, () => {
    test(
      `it should properly fix merge conflicts`,
      makeTemporaryEnv(
        {},
        async ({path, run, source}) => {
          await execFile(`git`, [`init`], {cwd: path});

          await run(`install`);
          await writeJson(`${path}/package.json`, {dependencies:{[`no-deps`]: `*`}});

          await execFile(`git`, [`add`, `-A`], {cwd: path});
          await execFile(`git`, [`commit`, `-a`, `-m`, `my-commit`], {cwd: path});

          await execFile(`git`, [`checkout`, `master`], {cwd: path});
          await execFile(`git`, [`checkout`, `-b`, `1.0.0`], {cwd: path});
          await run(`set`, `resolution`, `no-deps@npm:*`, `1.0.0`);
          await execFile(`git`, [`add`, `-A`], {cwd: path});
          await execFile(`git`, [`commit`, `-a`, `-m`, `commit-1.0.0`], {cwd: path});

          await execFile(`git`, [`checkout`, `master`], {cwd: path});
          await execFile(`git`, [`checkout`, `-b`, `2.0.0`], {cwd: path});
          await run(`set`, `resolution`, `no-deps@npm:*`, `2.0.0`);
          await execFile(`git`, [`add`, `-A`], {cwd: path});
          await execFile(`git`, [`commit`, `-a`, `-m`, `commit-2.0.0`], {cwd: path});

          await execFile(`git`, [`checkout`, `master`], {cwd: path});
          await execFile(`git`, [`merge`, `1.0.0`], {cwd: path});

          await expect(execFile(`git`, [`merge`, `2.0.0`], {cwd: path})).rejects.toThrow(/CONFLICT/);

          await expect(readFile(`${path}/yarn.lock`, `utf8`)).resolves.toMatchSnapshot();
          await expect(run(`install`)).resolves.toMatchSnapshot();
        },
      ),
    );
  });
});
