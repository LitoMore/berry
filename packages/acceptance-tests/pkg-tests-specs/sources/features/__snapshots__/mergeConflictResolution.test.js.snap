// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features Merge Conflict Resolution it should properly fix merge conflicts 1`] = `
"# This file is generated by running \\"berry install\\" inside your project.
# Manual changes might be lost - proceed with caution!

__metadata:
  version: 3

\\"no-deps@npm:*\\":
<<<<<<< HEAD
  version: 1.0.0
  resolution: \\"no-deps@npm:1.0.0\\"
  checksum: 25f16c3cf533d2bcee1aaeeb825bccee290dfb13e60442df735bb3521c95087f8b6a77accad79f79f3a51d60c6ed222b430a73edb4cf4306375e9329d9ba70f9
=======
  version: 2.0.0
  resolution: \\"no-deps@npm:2.0.0\\"
  checksum: 8ed1476baacbd817a052b42c19b5662bd2335c7b1dfc17cbee2b0dcf026d534c12631a9484865d3e84c68cf5ac72c112f506f5d5e058b5c5e21307ec223927c7
>>>>>>> 2.0.0
  languageName: node
  linkType: hard

\\"root-workspace-988eec@workspace:.\\":
  version: 0.0.0-use.local
  resolution: \\"root-workspace-988eec@workspace:.\\"
  dependencies:
    no-deps: \\"npm:*\\"
  languageName: unknown
  linkType: soft
"
`;

exports[`Features Merge Conflict Resolution it should properly fix merge conflicts 2`] = `
Object {
  "code": 0,
  "stderr": "",
  "stdout": "‚û§ YN0048: Automatically fixed merge conflicts üëç
‚û§ YN0000: ‚îå Resolution step
‚û§ YN0000: ‚îî Completed
‚û§ YN0000: ‚îå Fetch step
‚û§ YN0000: ‚îî Completed
‚û§ YN0000: ‚îå Link step
‚û§ YN0000: ‚îî Completed
‚û§ YN0000: Done
",
}
`;
